### PR TITLE
67 create recipe ziheng

### LINKED ISSUE #67: [IMPLEMENT] Create custom recipe page
# Feature Name

 Create custom recipe page

# Additional Context/help

Using the structure of the Detailed Recipe Page, populate each section with user-input text boxes. Save those into localStorage following the structure. 

Spoontacular has a function where you can scrape a recipe from any url and returns an identical json object from the API. Separate issue will be made for adding from urls.

# User Story

As a user, I would like to be able to create a custom recipe for my Grandma's Chocolate Chip Cookies and store it in my lists. 

*What a user would want to do with the product.*

### PR DESCRIPTION
**Pull request by hzhfred**
## New Changes
- add createRecipe to index.html under a section
- createRecipe.css in style
- createRecipe.js in js
- user can create recipe after clicking create button. (it will save the data to local storage under "created" category, it's also saved under "created" in saved list)

## Checklist
- [x] I have requested 2+ code reviewers
- [x] I have reviewed and commented my own code.
- [x] My code follows the style guidelines of this project.
- [ ] I have added new tests as necessary.
- [ ] All new and existing tests pass successfully.

### CODE REVIEW

File: source/styles/createRecipe.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+.ingredient-title{
+    padding-top: 10px;
+}
+
+.ingredient-input-list{
+    height: 180px;
+    overflow-y: scroll;
+    border-radius: 10px;
```
Comment (farisashai) [inline source/styles/createRecipe.css]: make sure you run `npm run lintcss` on your branch and fix any errors before we merge it

File: source/js/createRecipe.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
+import * as storageFuncs from './storage/storage.js';
+function init() {
```
Comment (farisashai) [inline source/js/createRecipe.js]: i think the function needs to be async?

File: source/index.html [hunk 3]
```diff
@@ -1070,6 +1084,169 @@ <h1>World’s Greatest Lasagna Roll Ups</h1>
     <section class="recipe-info hidden">
       
     </section>
+
+    <section class="create-recipe-page shown">
```
Comment (farisashai) [inline source/index.html]: does the class "shown" do anything?

**Comment by farisashai** [comment]
also, just as more design-related feedback
- for the ingredients list, i would place the name column first before amount and unit.
- For prep time and cook time, put "(minutes)" next to the label because the input field doesn't clarify that anywhere.
- Restrict the rating range from 0 to 5. i think you just write something like`<input min="0" max="5">`
- the bottom needs a margin after the create recipe button (match it to be the same size as the top margin)
- maybe remove exclamation points on buttons lol
- can we make the box border around the list of ingredient amounts invisible?

you should ask antonia or jon if they agree with these points or have any other feedback since this page was very new. everything else looks really nice though good job making it so fast 👍

File: source/index.html (hunk 3 above)
Comment (hzhfred) [reply source/index.html]: removed it

**Review by farisashai — APPROVED** [review]
looks good!

**Pull request merged**